### PR TITLE
feat: add block author example

### LINKED ISSUE #500: example deducing block author
Not exactly sure how this works. Something along the lines of...

- Read the latest block's header
- Extract digest logs
- Read latest Babe pallet authorities (could be different consensus though...)
- Iterate over authorities, decoding...
- ...

See [PJS equivalent](https://polkadot.js.org/docs/api/cookbook/blocks#how-do-i-extract-the-block-author)

### PR DESCRIPTION
**Pull request by ryanleecode**
Resolves #500

### CODE REVIEW

File: effects/rpc_known_methods.ts [hunk 1]
```diff
@@ -31,6 +31,7 @@ export namespace chain {
   export const getBlockHash = rpcCall<[height?: known.ListOrValue<known.NumberOrHex>], U.HexHash>(
     "chain_getBlockHash",
   )
+  export const getBlockHeader = rpcCall<[hash?: U.HexHash], known.Header>("chain_getHeader")
```
Comment (tjjfvi) [inline effects/rpc_known_methods.ts:34]:
```suggestion
  export const getHeader = rpcCall<[hash?: U.HexHash], known.Header>("chain_getHeader")
```

Comment (ryanleecode) [reply effects/rpc_known_methods.ts]: :heavy_check_mark:

File: examples/block_author.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import { $preDigest } from "http://localhost:5646/@local/proxy/wss:rpc.polkadot.io/types/sp_consensus_babe/digests.ts"
+import {
+  $digestItem,
+  DigestItem,
+} from "http://localhost:5646/@local/proxy/wss:rpc.polkadot.io/types/sp_runtime/generic/digest.ts"
+import * as U from "http://localhost:5646/@local/util/mod.ts"
+
+const blockHeader = C.chain.getHeader(C.polkadot)()
+const header = U.throwIfError(await blockHeader.run())
+const digestedLogs = header.digest.logs.map(C.hex.decode).map($digestItem.decode.bind($digestItem))
```
Comment (tjjfvi) [inline examples/block_author.ts:15]:
```suggestion
const digestLogs = header.digest.logs.map(x => $digestItem.decode(C.hex.decode(x)))
```

File: examples/block_author.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+function preRuntimeGuard(digestItem: DigestItem): digestItem is DigestItem.PreRuntime {
```
Comment (tjjfvi) [inline examples/block_author.ts:17]:
```suggestion
function isPreRuntime(digestItem: DigestItem): digestItem is DigestItem.PreRuntime {
```

File: examples/block_author.ts [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  return digestItem.type === "PreRuntime"
+}
+
+const digest = digestedLogs.find(preRuntimeGuard)
+if (!digest) {
+  throw new Error("Missing PreRuntime log")
+}
+
+const utf8decoder = new TextDecoder()
+const consensusEngineId = utf8decoder.decode(digest.value[0])
```
Comment (tjjfvi) [inline examples/block_author.ts:27]:
```suggestion
const consensusEngineId = new TextDecoder().decode(digest.value[0])
```

File: examples/block_author.ts (hunk 2 above)
Comment (ryanleecode) [reply examples/block_author.ts:15]: easier to read than map chaining eh? its what i had originally

Comment (tjjfvi) [reply examples/block_author.ts:15]: Easier to read and more efficient ;)

**Review by tjjfvi — APPROVED** [review]

**Pull request merged**